### PR TITLE
Moving all integration tests off of createTestServer

### PR DESCRIPTION
**Pull request by alyssawilk**
Moves all envoy tests off of createTestServer.
Sadly we can't yet move the old style json file creation to legacy_json_test.cc since I need to fix 
https://github.com/envoyproxy/envoy-filter-example/blob/master/echo2_integration_test.cc
first.

This includes extending the config utility to handle Envoy starting up with no listeners, better support for non-HTTP tests, and the reusable "base filter" yaml requested on the last v2 PR.

*Risk Level*: Low (test only)
*Testing*: n/a
*Docs Changes*: n/a
*Release Notes*: n/a

### CODE REVIEW

File: test/integration/integration.h [hunk 1]
```diff
@@ -116,7 +116,6 @@ class BaseIntegrationTest : Logger::Loggable<Logger::Id::testing> {
 
   void SetUp();
 
-  // TODO(alyssawilk) port the rest of the tests to v2 and make this non-virtual.
```
Comment (alyssawilk) [inline test/integration/integration.h]: Admittedly this is still virtual, but now subclasses which are doing any real work are required to call it, as validated in lookupPort, so I claim the TODO is done in spirit :-)

**Comment by alyssawilk** [comment]
Oh, I hadn't discovered the envoy-filter-example repo yet.    Phooey.  OK, I guess I'll back out the testServer change, go fix that test, and re-remove it.  

The v2 fun never ends!

**Comment by mattklein123** [comment]
In general LGTM, LMK when updated to work w/ envoy filter example. Doing this in two stages would be helpful as we have to port some internal Lyft integration tests also.

**Comment by alyssawilk** [comment]
I think this last upload (removing the ASSERT(initialized_)) has it.

I can leave the createTestServer in with comment-warnings for the long term or you folks can have an OldSchoolTestBase which implements it for your internal tests - whatever you think is best.

**Comment by mattklein123** [comment]
> I can leave the createTestServer in with comment-warnings for the long term or you folks can have an OldSchoolTestBase which implements it for your internal tests - whatever you think is best.

Yeah it's fine either way. We can get our tests updated. Does envoy-filter-example need to be updated?

File: test/integration/integration.h [hunk 2]
```diff
@@ -116,8 +116,8 @@ class BaseIntegrationTest : Logger::Loggable<Logger::Id::testing> {
...
   // Initialize the basic proto configuration, create fake upstreams, and start Envoy.
+  // // TODO(alyssawilk) port the rest of the tests to v2 and make initialized_ private.
```
Comment (mattklein123) [inline test/integration/integration.h]: errant '//' ?

File: test/integration/integration.cc [hunk 3]
```diff
@@ -269,10 +269,12 @@ void BaseIntegrationTest::registerTestServerPorts(const std::vector<std::string>
 void BaseIntegrationTest::createGeneratedApiTestServer(const std::string& bootstrap_path,
                                                        const std::vector<std::string>& port_names) {
   test_server_ = IntegrationTestServer::create(bootstrap_path, version_);
-  // Need to ensure we have an LDS update before invoking registerTestServerPorts() below that
-  // needs to know about the bound listener ports.
-  test_server_->waitForCounterGe("listener_manager.listener_create_success", 1);
-  registerTestServerPorts(port_names);
+  if (config_helper_.bootstrap().static_resources().listeners_size() > 0) {
+    // Need to ensure we have an LDS update before invoking registerTestServerPorts() below that
```
Comment (mattklein123) [inline test/integration/integration.cc]: I don't think this is waiting for LDS update? Just waiting for listener create? (I think old not correct comment)

File: test/integration/echo_integration_test.cc [hunk 4]
```diff
@@ -3,19 +3,36 @@
...
+          descriptors: [{"key": "foo", "value": "bar"}]
+      filters:
+        name: envoy.echo
+        config:
+      )EOF";
+  }
+
   /**
-   * Initializer for an individual test.
+   *  Initializer for an individual test.
```
Comment (mattklein123) [inline test/integration/echo_integration_test.cc]: errant ' ' ?

File: test/integration/echo_integration_test.cc [hunk 5]
```diff
@@ -3,19 +3,36 @@
...
+
 class EchoIntegrationTest : public BaseIntegrationTest,
                             public testing::TestWithParam<Network::Address::IpVersion> {
 public:
-  EchoIntegrationTest() : BaseIntegrationTest(GetParam()) {}
+  EchoIntegrationTest() : BaseIntegrationTest(GetParam(), echo_config) {}
+
+  // Called once by the gtest framework before any EchoIntegrationTests are run.
+  static void SetUpTestCase() {
+    echo_config = ConfigHelper::BASE_CONFIG + R"EOF(
```
Comment (mattklein123) [inline test/integration/echo_integration_test.cc]: nit: Could this just be a const member variable?

**Comment by htuch** [comment]
Yeah, there is both a network and HTTP filter example in https://github.com/envoyproxy/envoy-filter-example. The HTTP filter example is now v2 based, the network filter example needs updating.

File: test/integration/echo_integration_test.cc (hunk 5 above)
Comment (alyssawilk) [reply test/integration/echo_integration_test.cc]: I don't think so - when I tried it that way it segfaulted as it created the local echo_config before BASE_CONFIG was initialized.  Could have been PEBAC of some sort - I can give it another shot if you think it ought to work

File: test/integration/echo_integration_test.cc (hunk 4 above)
Comment (alyssawilk) [reply test/integration/echo_integration_test.cc]: I loves me my whitespace :-P

(no, this is flat out wrong, fixed locally)

File: test/integration/echo_integration_test.cc (hunk 5 above)
Comment (mattklein123) [reply test/integration/echo_integration_test.cc]: It seems like it should work but it's not a big deal. I wouldn't worry about it.

**Pull request merged**